pulsar-functions/runtime/src/main/java/org/apache/pulsar/functions/auth/NoOpFunctionAuthProvider.java

```diff
@@ -0,0 +1,43 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pulsar.functions.auth;
+
+import org.apache.pulsar.broker.authentication.AuthenticationDataSource;
+import org.apache.pulsar.functions.instance.AuthenticationConfig;
+import org.apache.pulsar.functions.proto.Function;
+
+public class NoOpFunctionAuthProvider implements FunctionAuthProvider{
+    @Override
+    public void configureAuthenticationConfig(AuthenticationConfig authConfig, Function.FunctionAuthenticationSpec
+            functionAuthenticationSpec) {
+
+    }
+
+    @Override
+    public Function.FunctionAuthenticationSpec cacheAuthData(String tenant, String namespace, String name,
+                                                             AuthenticationDataSource authenticationDataSource)
+            throws Exception {
+        return null;
```

Comment:
do we really need return a null here? or shall returning a spec with a `data`?

Reply:
I don't quite follow

Reply:
I mean returning null usually results in NPE if you don't take care of it at the callers.

Reply:
I use null to gate whether we should do anything or not:

https://github.com/apache/pulsar/pull/3735/files#diff-d906ae3d0366fbf1866c0334f075b32eR394

Reply:
From an API perspective, it's better to use `Optional<X>` whenever the returned object might not be applicable.

Reply:
ok we can change this method to return Optional<Function.FunctionAuthenticationSpe>.  Is that what you would recommend?